### PR TITLE
Forecasting Air Temperature in California using ResCNN model

### PR DESCRIPTION
**Pull request by anushriyaj**
Forecasting Air Temperature in California Notebook.

-----

# Checklist

Please go through each entry in the below checklist and mark an 'X' if that condition has been met. Every entry should be marked with an 'X' to be get the Pull Request approved.

- [ ] All `import`s are in the first cell? First block of imports are standard libraries, second block are 3rd party libraries, third block are all `arcgis` imports? Note that in some cases, for samples, it is a good idea to keep the imports next to where they are used, particularly for uncommonly used features that we want to highlight.
- [ ] All `GIS` object instantiations are one of the following?
    - `gis = GIS()`
    - `gis = GIS('home')`
    - `gis = GIS(profile="your_online_portal")`
    - `gis = GIS('https://pythonapi.playground.esri.com/portal')`
    - `gis = GIS(profile="your_enterprise_portal")`
- [ ] If this notebook requires setup or teardown, did you add the appropriate code to `./misc/setup.py` and/or `./misc/teardown.py`?
- [ ] If this notebook references any portal items that need to be staged on AGOL/Python API playground, did you coordinate with a Python API team member to stage the item the correct way with the api\_data\_owner user?
- [ ] Code refactored & split out across multiple cells, useful comments?
- [ ] Consistent voice/tense/narrative style? Thoroughly checked for typos?
- [ ] All images used like `` instead of ``? All map widgets contain a static image preview? (Call `mapview_inst.take_screenshot()` to do so)
- [ ] All file paths are constructed in an OS-agnostic fashion with `os.path.join()`? (Instead of `r"\foo\bar"`, `os.path.join(os.path.sep, "foo", "bar")`, etc.)
- [ ] **IF YOU WANT THIS SAMPLE TO BE DISPLAYED ON THE DEVELOPERS.ARCGIS.COM WEBSITE**, ping @ DavidJVitale so he can add it to the list for the next deploy

### CODE REVIEW

**Comment by priyankatuteja** [comment]
@moonlanderr Could you please help in review of this sample

**Comment by moonlanderr** [comment]
@anushriyaj please add the suggested changes

File: samples/04_gis_analysts_data_scientists/forecasting_air_temperature_in_california.ipynb [hunk 1]
```diff
@@ -0,0 +1,2703 @@
+{
```
Comment (moonlanderr) [inline samples/04_gis_analysts_data_scientists/forecasting_air_temperature_in_california.ipynb:1]: <p>The TimeSeriesModel from arcgis.learn includes backbones such as InceptionTime, ResCNN, <strong>ResNet</strong> and FCN which does not need fine-tuning of multiple hyperparameters before fitting the model. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1119/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (moonlanderr) [inline samples/04_gis_analysts_data_scientists/forecasting_air_temperature_in_california.ipynb:1]: <p>chart is not visible, convert it to base64 </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1119/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (moonlanderr) [inline samples/04_gis_analysts_data_scientists/forecasting_air_temperature_in_california.ipynb:1]: <p>The data used in this sample study is a multivariate monthly time series dataset recorded at a ground station in the Fresno Yosemite International Airport, California, USA. It ranges from January 1948 to November 2015.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1119/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (moonlanderr) [inline samples/04_gis_analysts_data_scientists/forecasting_air_temperature_in_california.ipynb:1]: <p>map not visible,</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1119/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (moonlanderr) [inline samples/04_gis_analysts_data_scientists/forecasting_air_temperature_in_california.ipynb:1]: <p>put these in one cell for Visualize as pandas dataframe</p><p>climate_data = data_table.tables[0]</p><p>climate_df = climate_data.query().sdf</p><p>climate_df.head()</p><p> </p><p>Also can you explain the dataset, defining the column names, what they represent</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1119/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (moonlanderr) [inline samples/04_gis_analysts_data_scientists/forecasting_air_temperature_in_california.ipynb:1]: <p>Next, the dataset is prepared wherein the variables of station, possible sunshine, snow cover, maximum temperature, minimum temperature, total sunshine, and peak wind gust speed are dropped, and then the dataset is chosen from 1987 onwards, to avoid missing values.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1119/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (moonlanderr) [inline samples/04_gis_analysts_data_scientists/forecasting_air_temperature_in_california.ipynb:1]: <p>can you explain the variable predict and the predictors to be used</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1119/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (moonlanderr) [inline samples/04_gis_analysts_data_scientists/forecasting_air_temperature_in_california.ipynb:1]: <h3>Data types of time series variables</h3>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1119/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Comment (moonlanderr) [inline samples/04_gis_analysts_data_scientists/forecasting_air_temperature_in_california.ipynb:1]: <p>Here in the original dataset the <code>variable predict</code> column of Average Temperature (<code>TAVG</code>) is allocated with NaNs, for the forecasting period of 2014-2015. This format is required for the&nbsp;model.predict()&nbsp;function in time series analysis, which will fill up the NaN values with forecasted temperatures.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1119/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

Comment (moonlanderr) [inline samples/04_gis_analysts_data_scientists/forecasting_air_temperature_in_california.ipynb:1]: <p>print out the final prepared data showing the tail where the nan values are</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1119/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Comment (moonlanderr) [inline samples/04_gis_analysts_data_scientists/forecasting_air_temperature_in_california.ipynb:1]: <p>combine the above two lines in the same cell</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1119/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (moonlanderr) [inline samples/04_gis_analysts_data_scientists/forecasting_air_temperature_in_california.ipynb:1]: <p>define precisely what each of the following parameters are using in the current model, rather than just describing the parameters, </p><p> </p><p><code>The primary input parameters required for the tool are: </code></p><ul><li><code>input_features&nbsp;: non spatial dataframe, feature layer, or spatial dataframe containing the primary dataset and the explanatory variables, if there are any</code></li><li><code>variable_predict&nbsp;: field name containing the y-variable to be forecasted from the input feature layer/dataframe</code></li><li><code>explanatory_variables&nbsp;: list of the field names as 2-sized tuples containing the explanatory variables as mentioned above. Since there are none in this example, it is not required here</code></li><li><code>index_field&nbsp;: field name containing the timestamp</code></li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1119/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (moonlanderr) [inline samples/04_gis_analysts_data_scientists/forecasting_air_temperature_in_california.ipynb:1]: <p>The predicted vs the actual values by the trained model is printed for the training dataset</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1119/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='54'/>

Comment (moonlanderr) [inline samples/04_gis_analysts_data_scientists/forecasting_air_temperature_in_california.ipynb:1]: <p>During forecasting, the model uses dataset prepared above with NaN values as input with <code>prediction_type</code> as <code>dataframe</code>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1119/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='57'/>

Comment (moonlanderr) [inline samples/04_gis_analysts_data_scientists/forecasting_air_temperature_in_california.ipynb:1]: <p>Check the correct color of the lines for forecast and actual</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1119/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='66'/>

Comment (moonlanderr) [inline samples/04_gis_analysts_data_scientists/forecasting_air_temperature_in_california.ipynb:1]: <p>using Deep learning TimeSeriesModel from the arcgis.learn and forecasted monthly <strong>Air temperature</strong> for a station in California. The model was trained with 25 years (1987-2013) of data and forecasted for a period of 2 years (2014-2015) with high accuracy. The dependent variables were wind speed and precipitation.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1119/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='69'/>

**Comment by anushriyaj** [comment]
@moonlanderr Thank you for reviewing, I have incorporated your comments. Please Check once.

**Comment by AtmaMani** [comment]
@BP-Ent could you please review this PR?

**Review by BP-Ent — CHANGES_REQUESTED** [review]
Suggested changes made on Reviewnb

**Comment by anushriyaj** [comment]
<p>Right it should be 23 instead of 12, Corrected.</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1119/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by anushriyaj** [comment]
<p>Yes, those are explanatory variables. Corrected in the sentence.</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1119/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by anushriyaj** [comment]
@BP-Ent Thank you for reviewing, I have incorporated all the suggestions. Please Check.

**Review by BP-Ent — CHANGES_REQUESTED** [review]
Some more minor changes suggested on reviewNb. Should be good to go after that!

**Comment by anushriyaj** [comment]
Edits done, Thanks @BP-Ent. Now it is good to go @AtmaMani @priyankatuteja

**Comment by AtmaMani** [comment]
@anushriyaj this is great article, well written. Thank you. Please update the title and filename per the recommendation.
@moonlanderr @BP-Ent thank you for the detailed review.

**Comment by anushriyaj** [comment]
@AtmaMani Thank you. I have changed the title and file name as suggested.

**Review by AtmaMani — CHANGES_REQUESTED** [review]
@anushriyaj file names should always be in full lowercase with either hyphen or underscore to separate words. Please change `ResCNN` to `rescnn` in the filename

**Comment by anushriyaj** [comment]
@AtmaMani Right, Thank you I have renamed it.

**Review by AtmaMani — APPROVED** [review]

**Comment by araedavis** [comment]
@AtmaMani I was just running `afd-python-api-doc` locally and our Gatsby plugin that sources the guide and sample notebooks is throwing an error on this file - it looks like the notebook itself was deleted when the file was renamed.

**Comment by AtmaMani** [comment]
@araedavis thanks for catching this! Yes, the file has no content.

@anushriyaj could you make another PR to fix this issue? Looks like during the rename process git wiped the content. File renaming on windows OS and git can get challenging. I suggest moving the file to a different folder (outside this repo) making a commit, rename the file outside, bring it back into the repo, make a commit. Then push up your branch and make a PR

**Comment by mohi9282** [comment]
@anushriyaj - any updates on the changes requested?

**Comment by anushriyaj** [comment]
@mohi9282 - The notebook with asked changes is approved and merged with this new PR [https://github.com/Esri/arcgis-python-api/pull/1143]
cc @priyankatuteja
